build: various "unused function" in map_defs.h

### Problem

```
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:22:24: warning: unused function 'hash_String' [clang-diagnostic-unused-function]
static inline uint32_t hash_String(String s)
                       ^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:31:20: warning: unused function 'equal_String' [clang-diagnostic-unused-function]
static inline bool equal_String(String a, String b)
                   ^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:144:1: warning: unused function 'set_del_int' [clang-diagnostic-unused-function]
KEY_DECLS(int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:144:1: warning: unused function 'set_has_int' [clang-diagnostic-unused-function]
KEY_DECLS(int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:144:1: warning: unused function 'set_put_int' [clang-diagnostic-unused-function]
KEY_DECLS(int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:145:1: warning: unused function 'set_del_cstr_t' [clang-diagnostic-unused-function]
KEY_DECLS(cstr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:145:1: warning: unused function 'set_has_cstr_t' [clang-diagnostic-unused-function]
KEY_DECLS(cstr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:145:1: warning: unused function 'set_put_cstr_t' [clang-diagnostic-unused-function]
KEY_DECLS(cstr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:146:1: warning: unused function 'set_del_ptr_t' [clang-diagnostic-unused-function]
KEY_DECLS(ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:146:1: warning: unused function 'set_has_ptr_t' [clang-diagnostic-unused-function]
KEY_DECLS(ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:146:1: warning: unused function 'set_put_ptr_t' [clang-diagnostic-unused-function]
KEY_DECLS(ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:147:1: warning: unused function 'set_del_uint64_t' [clang-diagnostic-unused-function]
KEY_DECLS(uint64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:147:1: warning: unused function 'set_has_uint64_t' [clang-diagnostic-unused-function]
KEY_DECLS(uint64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:147:1: warning: unused function 'set_put_uint64_t' [clang-diagnostic-unused-function]
KEY_DECLS(uint64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:148:1: warning: unused function 'set_del_int64_t' [clang-diagnostic-unused-function]
KEY_DECLS(int64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:148:1: warning: unused function 'set_has_int64_t' [clang-diagnostic-unused-function]
KEY_DECLS(int64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:148:1: warning: unused function 'set_put_int64_t' [clang-diagnostic-unused-function]
KEY_DECLS(int64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:149:1: warning: unused function 'set_del_uint32_t' [clang-diagnostic-unused-function]
KEY_DECLS(uint32_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:149:1: warning: unused function 'set_has_uint32_t' [clang-diagnostic-unused-function]
KEY_DECLS(uint32_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:149:1: warning: unused function 'set_put_uint32_t' [clang-diagnostic-unused-function]
KEY_DECLS(uint32_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:150:1: warning: unused function 'set_del_String' [clang-diagnostic-unused-function]
KEY_DECLS(String)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:150:1: warning: unused function 'set_has_String' [clang-diagnostic-unused-function]
KEY_DECLS(String)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:150:1: warning: unused function 'set_put_String' [clang-diagnostic-unused-function]
KEY_DECLS(String)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:151:1: warning: unused function 'set_del_HlEntry' [clang-diagnostic-unused-function]
KEY_DECLS(HlEntry)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:151:1: warning: unused function 'set_has_HlEntry' [clang-diagnostic-unused-function]
KEY_DECLS(HlEntry)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:151:1: warning: unused function 'set_put_HlEntry' [clang-diagnostic-unused-function]
KEY_DECLS(HlEntry)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:152:1: warning: unused function 'set_del_ColorKey' [clang-diagnostic-unused-function]
KEY_DECLS(ColorKey)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:108:19: note: expanded from macro 'KEY_DECLS'
  static inline T set_del_##T(Set(T) *set, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:152:1: warning: unused function 'set_has_ColorKey' [clang-diagnostic-unused-function]
KEY_DECLS(ColorKey)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:113:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_has_##T(Set(T) *set, T key) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:152:1: warning: unused function 'set_put_ColorKey' [clang-diagnostic-unused-function]
KEY_DECLS(ColorKey)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:100:22: note: expanded from macro 'KEY_DECLS'
  static inline bool set_put_##T(Set(T) *set, T key, T **key_alloc) { \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:154:1: warning: unused function 'map_get_intint' [clang-diagnostic-unused-function]
MAP_DECLS(int, int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:154:1: warning: unused function 'map_put_intint' [clang-diagnostic-unused-function]
MAP_DECLS(int, int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:155:1: warning: unused function 'map_get_intptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(int, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:155:1: warning: unused function 'map_put_intptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(int, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:156:1: warning: unused function 'map_get_cstr_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(cstr_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:156:1: warning: unused function 'map_put_cstr_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(cstr_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:157:1: warning: unused function 'map_get_cstr_tint' [clang-diagnostic-unused-function]
MAP_DECLS(cstr_t, int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:157:1: warning: unused function 'map_put_cstr_tint' [clang-diagnostic-unused-function]
MAP_DECLS(cstr_t, int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:158:1: warning: unused function 'map_get_ptr_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(ptr_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:158:1: warning: unused function 'map_put_ptr_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(ptr_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:159:1: warning: unused function 'map_get_uint32_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint32_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:159:1: warning: unused function 'map_put_uint32_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint32_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:160:1: warning: unused function 'map_get_uint64_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint64_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:160:1: warning: unused function 'map_put_uint64_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint64_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:161:1: warning: unused function 'map_get_uint64_tssize_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint64_t, ssize_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:161:1: warning: unused function 'map_put_uint64_tssize_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint64_t, ssize_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:162:1: warning: unused function 'map_get_uint64_tuint64_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint64_t, uint64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:162:1: warning: unused function 'map_put_uint64_tuint64_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint64_t, uint64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:163:1: warning: unused function 'map_get_int64_tint64_t' [clang-diagnostic-unused-function]
MAP_DECLS(int64_t, int64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:163:1: warning: unused function 'map_put_int64_tint64_t' [clang-diagnostic-unused-function]
MAP_DECLS(int64_t, int64_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:164:1: warning: unused function 'map_get_int64_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(int64_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:164:1: warning: unused function 'map_put_int64_tptr_t' [clang-diagnostic-unused-function]
MAP_DECLS(int64_t, ptr_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:165:1: warning: unused function 'map_get_uint32_tuint32_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint32_t, uint32_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:165:1: warning: unused function 'map_put_uint32_tuint32_t' [clang-diagnostic-unused-function]
MAP_DECLS(uint32_t, uint32_t)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:166:1: warning: unused function 'map_get_Stringint' [clang-diagnostic-unused-function]
MAP_DECLS(String, int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:166:1: warning: unused function 'map_put_Stringint' [clang-diagnostic-unused-function]
MAP_DECLS(String, int)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:167:1: warning: unused function 'map_get_intString' [clang-diagnostic-unused-function]
MAP_DECLS(int, String)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:167:1: warning: unused function 'map_put_intString' [clang-diagnostic-unused-function]
MAP_DECLS(int, String)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
                     ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:168:1: warning: unused function 'map_get_ColorKeyColorItem' [clang-diagnostic-unused-function]
MAP_DECLS(ColorKey, ColorItem)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:125:19: note: expanded from macro 'MAP_DECLS'
  static inline U map_get_##T##U(Map(T, U) *map, T key) \
                  ^
note: expanded from here
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:168:1: warning: unused function 'map_put_ColorKeyColorItem' [clang-diagnostic-unused-function]
MAP_DECLS(ColorKey, ColorItem)
^
/home/runner/work/neovim/neovim/src/nvim/map_defs.h:132:22: note: expanded from macro 'MAP_DECLS'
  static inline void map_put_##T##U(Map(T, U) *map, T key, U value) \
```

### Expected behavior

Some of these are unavoidable, but some of the `MAP_DECLS` may be able to be deleted: https://github.com/neovim/neovim/blob/5fa88a7ae13effc02d42070313d3082bf98e22a7/src/nvim/map_defs.h#L154-L168

--- a/src/nvim/map.c
+++ b/src/nvim/map.c
@@ -102,9 +102,6 @@ void mh_clear(MapHash *h)
 
 #define KEY_NAME(x) x##int
 #include "nvim/map_key_impl.c.h"
-#define VAL_NAME(x) quasiquote(x, int)
-#include "nvim/map_value_impl.c.h"
-#undef VAL_NAME
 #define VAL_NAME(x) quasiquote(x, ptr_t)
 #include "nvim/map_value_impl.c.h"
 #undef VAL_NAME
@@ -150,12 +147,6 @@ void mh_clear(MapHash *h)
 #define KEY_NAME(x) x##uint64_t
 #include "nvim/map_key_impl.c.h"
 #define VAL_NAME(x) quasiquote(x, ptr_t)
-#include "nvim/map_value_impl.c.h"
-#undef VAL_NAME
-#define VAL_NAME(x) quasiquote(x, ssize_t)
-#include "nvim/map_value_impl.c.h"
-#undef VAL_NAME
-#define VAL_NAME(x) quasiquote(x, uint64_t)
 #include "nvim/map_value_impl.c.h"
 #undef VAL_NAME
 #define VAL_NAME(x) quasiquote(x, int)

--- a/src/nvim/map_defs.h
+++ b/src/nvim/map_defs.h
@@ -43,9 +43,7 @@ static inline bool equal_String(String a, String b)
 
 static const int value_init_int = 0;
 static const ptr_t value_init_ptr_t = NULL;
-static const ssize_t value_init_ssize_t = -1;
 static const uint32_t value_init_uint32_t = 0;
-static const uint64_t value_init_uint64_t = 0;
 static const int64_t value_init_int64_t = 0;
 static const String value_init_String = STRING_INIT;
 static const ColorItem value_init_ColorItem = COLOR_ITEM_INITIALIZER;
@@ -153,15 +151,12 @@ KEY_DECLS(String)
 KEY_DECLS(HlEntry)
 KEY_DECLS(ColorKey)
 
-MAP_DECLS(int, int)
 MAP_DECLS(int, ptr_t)
 MAP_DECLS(cstr_t, ptr_t)
 MAP_DECLS(cstr_t, int)
 MAP_DECLS(ptr_t, ptr_t)
 MAP_DECLS(uint32_t, ptr_t)
 MAP_DECLS(uint64_t, ptr_t)
-MAP_DECLS(uint64_t, ssize_t)
-MAP_DECLS(uint64_t, uint64_t)
 MAP_DECLS(uint64_t, int)
 MAP_DECLS(int64_t, int64_t)
 MAP_DECLS(int64_t, ptr_t)

